haproxy/Dockerfile

```diff
@@ -0,0 +1,77 @@
+#FROM ubuntu:jammy-20230425
+FROM ubuntu:focal-20230412 AS build
+
+ENV HAPROXY_BRANCH 2.6
+ENV HAPROXY_MINOR 2.6.10
+ENV HAPROXY_SHA256 e71b2cd9ca1043345f083a5225078ccf824dced2b5779d86f11fa4e88f451773
+ENV HAPROXY_SRC_URL http://www.haproxy.org/download
+
+ENV HAPROXY_UID haproxy
+ENV HAPROXY_GID haproxy
+ENV DEBIAN_FRONTEND noninteractive
+
+# if you use ubuntu:jammy, you must use "libssl3=3.0.2-0ubuntu1.9". but it is few case.
+# We choose "ubuntu:focal" and "libssl1.1" to get a stable operation.
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends procps zlib1g libssl1.1 "libpcre2-*" liblua5.3-0 libatomic1 tar curl socat ca-certificates && \
+    apt-get install -y --no-install-recommends gcc make libc6-dev libssl-dev libpcre3-dev zlib1g-dev liblua5.3-dev && \
+    curl -sfSL "${HAPROXY_SRC_URL}/${HAPROXY_BRANCH}/src/haproxy-${HAPROXY_MINOR}.tar.gz" -o haproxy.tar.gz && \
+    echo "$HAPROXY_SHA256 *haproxy.tar.gz" | sha256sum -c - && \
+    groupadd "$HAPROXY_GID" && \
+    useradd -g "$HAPROXY_GID" "$HAPROXY_UID" && \
+    mkdir -p /tmp/haproxy && \
+    tar -xzf haproxy.tar.gz -C /tmp/haproxy --strip-components=1 && \
+    rm -f haproxy.tar.gz && \
+    make -C /tmp/haproxy -j"$(nproc)" TARGET=linux-glibc CPU=generic USE_PCRE2=1 USE_PCRE2_JIT=1 USE_OPENSSL=1 \
+                            USE_TFO=1 USE_LINUX_TPROXY=1 USE_LUA=1 USE_GETADDRINFO=1 \
+                            USE_PROMEX=1 USE_SLZ=1 \
+                            all && \
+    make -C /tmp/haproxy TARGET=linux-glibc install-bin install-man # && \
+    ln -s /usr/local/sbin/haproxy /usr/sbin/haproxy && \
+    mkdir -p /var/lib/haproxy && \
+    chown "$HAPROXY_UID:$HAPROXY_GID" /var/lib/haproxy && \
+    mkdir -p /usr/local/etc/haproxy && \
+    ln -s /usr/local/etc/haproxy /etc/haproxy && \
+    cp -R /tmp/haproxy/examples/errorfiles /usr/local/etc/haproxy/errors && \
+    rm -rf /tmp/haproxy
+
+FROM ubuntu:focal-20230412
+LABEL maintainer="takara@cybozu.co.jp"
+LABEL name="HAProxy"
+LABEL release="Community Edition"
+LABEL vendor="HAProxy"
+LABEL version="2.6.10"
+LABEL org.opencontainers.image.source="https://github.com/cybozu/neco-containers/haproxy"
+
+ENV HAPROXY_BRANCH 2.6
+ENV HAPROXY_MINOR 2.6.10
+ENV HAPROXY_SHA256 e71b2cd9ca1043345f083a5225078ccf824dced2b5779d86f11fa4e88f451773
+ENV HAPROXY_SRC_URL http://www.haproxy.org/download
+
+ENV HAPROXY_UID haproxy
+ENV HAPROXY_GID haproxy
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN mkdir -p /var/lib/haproxy && \
+    mkdir -p /usr/local/etc/haproxy
+
+COPY --from=build /usr/local/sbin/haproxy /usr/sbin/haproxy
+COPY --from=build /tmp/haproxy/examples/errorfiles /usr/etc/haproxy/errors
+COPY docker-entrypoint.sh /
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends procps zlib1g libssl1.1 "libpcre2-*" liblua5.3-0 libatomic1 socat ca-certificates && \
```

Comment:
Are these packages required to run HAProxy?

Reply:
I remove HAPROXY_SHA256 and HAPROXY_SRC_URL.

Reply:
These packages required by Shell script for a Redis cluster. This HA-PROXY has responsibility to identify master redis node. 
To identify, the shell scirpt accesses Sentinel of redis. shell script supply to HA-Proxy by ConfigMap. The Shell scrip supply ArgoCD project.